source/Components/Button.cpp

```diff
@@ -0,0 +1,328 @@
+/*
+** Project FileShare, 2022
+**
+** Author Léo Lhuile
+**
+** Started on  Sun Feb 13 22:03:32 2022 Léo Lhuile
+** Last update Mon Aug 29 20:45:51 2022 Léo Lhuile
+**
+** Button.cpp : Implementation of Button class
+*/
+
+#include "Components/Button.hpp"
+#include "Debug.hpp"
+
+namespace FileShare::GUI::Components {
+    Button::Button(const char* typeName, bool initRenderer)
+        : tgui::BitmapButton(typeName, initRenderer)
+    {
+        for (const auto &component : this->m_backgroundComponent->getComponents()) {
+            auto imageComponent = std::dynamic_pointer_cast<tgui::priv::dev::ImageComponent>(component);
+            if (imageComponent) {
+                this->imageComponent = imageComponent;
+                break;
+            }
+        }
+
+        this->setSize(Size::Normal);
+        this->updateAligment();
+        this->updateStyle();
+        this->onClick(&Button::setFocused, this, false);
+    }
+
+    Button::~Button() {}
+
+    void Button::canBeActive(bool active, bool automatic)
+    {
+        if (active == this->isActive()) {
+            return;
+        }
+
+        if (active) {
+            this->activeSignalId = automatic ? this->onClick(&Button::toggleActive, this) : 1;
+        } else {
+            if (this->activeSignalId > 1) {
```

Comment:
What does the signalId 1 do ?

Reply:
Currently, it does nothing: I just use it to have a valid value in activeSignalId, and I use it to know if there was a previously connected signal or not (1 == no signal connected)

Reply:
But can't the `onClick` return signal_id `1` ?
I don't see any guarantees in the documentation about the range of ids, won't this misbehave if tgui sometimes returns `1` as id ?

Reply:
The ID is incremented with each new signal, and TGui uses some signals in its internal functioning. So it will never return an ID = 1